src/Components/DrillPage.js

```diff
@@ -1,25 +1,32 @@
 import React, { Component } from 'react';
-import { ScrollView, StyleSheet, View, Text, Button, ImageBackground, TouchableOpacity } from 'react-native';
+import { ScrollView, StyleSheet, View, Text, Image, ImageBackground, TouchableOpacity } from 'react-native';
 import { connect } from 'react-redux';
 
 class DrillPage extends React.Component {
+  _toggleFavorite() {
+    console.log('TOGGLE FAVORITE function ', this.state.drill);
```

Comment:
Here you cannot use `this.state.drill` because it had not been initialized (as you said in your message)

But you don't need to put the drill in the state juste to be able to call it here. We use the drills in the props to render the view. Maybe fetch it the same way ?
`this.props.route.params.drill`